npm/private/lifecycle/lifecycle-hooks.js

```diff
@@ -173,6 +187,8 @@ async function main(args) {
         pkgRoot: path.resolve(outputDir),
         rawConfig: {
             stdio: 'inherit',
+            platform: platform,
```

Comment:
Nice. I wasn't aware you could set these here. I tracked this down to ending up in `config` on the `lifecycle` call to `@pnpm/npm-lifecycle` and ending up the `env`. Could leave a brief comment here as to how it works?

Reply:
Links to docs or directly to code would be helpful for the next person who looks at this

Reply:
Added.